src/shaders/custom/post_process/ToneMapping.frag

```diff
@@ -49,8 +49,8 @@ void main() {
         // gamma correction 
         mapped = pow(mapped, vec3(1.0 / gamma));
     
-        if (hdrColor.a == 0.0)
-            ldrColor = vec4(u_clearColor.rgb, 1.0);
+        if (hdrColor.a < 1.0)
+            ldrColor = vec4(mapped * hdrColor.a + u_clearColor.rgb * (1.0 - hdrColor.a), 1.0);
         else
```

Comment:
I would remove the whole `else` branch and only keep the `ldrColor` evaluation (and maybe clamp the `hdrColor.a` value).